test/src/main/java/org/apache/accumulo/test/functional/ExitCodesIT.java

```diff
@@ -0,0 +1,334 @@
+    // Start MiniCluster so that getCluster() does not
+    // return null,
+    SharedMiniClusterBase.startMiniCluster();
+    getCluster().getClusterControl().stop(ServerType.GARBAGE_COLLECTOR);
+  }
+
+  @AfterAll
+  public static void afterTests() {
+    SharedMiniClusterBase.stopMiniCluster();
+  }
+
+  // Junit doesn't support more than one parameterized
+  // argument to a test method. We need to generate
+  // the arguments here.
+  static Stream<Arguments> generateWorkerProcessArguments() {
+    List<Arguments> args = new ArrayList<>();
+    for (ServerType st : ServerType.values()) {
+      if (st == ServerType.COMPACTOR || st == ServerType.SCAN_SERVER
+          || st == ServerType.TABLET_SERVER) {
+        for (TerminalBehavior tb : TerminalBehavior.values()) {
+          args.add(Arguments.of(st, tb));
+        }
+      }
+    }
+    return args.stream();
+  }
+
+  @ParameterizedTest
+  @MethodSource("generateWorkerProcessArguments")
+  public void testWorkerProcesses(ServerType server, TerminalBehavior behavior) throws Exception {
+    Map<String,String> properties = new HashMap<>();
+    properties.put(PROXY_METHOD_BEHAVIOR, behavior.name());
+    getCluster().getConfig().setSystemProperties(properties);
+    Class<?> serverClass = null;
+    switch (server) {
+      case COMPACTOR:
+        serverClass = ExitCompactor.class;
+        break;
+      case SCAN_SERVER:
+        serverClass = ExitScanServer.class;
+        break;
+      case TABLET_SERVER:
+        serverClass = ExitTabletServer.class;
+        break;
+      case GARBAGE_COLLECTOR:
+      case MANAGER:
+      case MONITOR:
+      case ZOOKEEPER:
+      default:
+        throw new IllegalArgumentException("Unhandled type");
+    }
+    ProcessInfo pi = getCluster()._exec(serverClass, server, Map.of(), new String[] {});
+    Wait.waitFor(() -> !pi.getProcess().isAlive(), 120_000);
+    int exitValue = pi.getProcess().exitValue();
+    assertEquals(behavior == TerminalBehavior.SHUTDOWN ? 0 : 1, exitValue);
+  }
+
+  @ParameterizedTest
+  @EnumSource
+  public void testGarbageCollector(TerminalBehavior behavior) throws Exception {
+    Map<String,String> properties = new HashMap<>();
+    properties.put(PROXY_METHOD_BEHAVIOR, behavior.name());
+    getCluster().getConfig().setSystemProperties(properties);
+    ProcessInfo pi =
+        getCluster()._exec(ExitGC.class, ServerType.GARBAGE_COLLECTOR, Map.of(), new String[] {});
+    if (behavior == TerminalBehavior.SHUTDOWN) {
+      Wait.waitFor(() -> !pi.getProcess().isAlive(), 120_000);
+      int exitValue = pi.getProcess().exitValue();
+      assertEquals(0, exitValue);
+    } else if (behavior == TerminalBehavior.EXCEPTION) {
+      // GarbageCollector logs exceptions and keeps going.
+      // We need to let this time out and then
+      // terminate the process.
+      IllegalStateException ise = assertThrows(IllegalStateException.class,
+          () -> Wait.waitFor(() -> !pi.getProcess().isAlive(), 120_000));
```

Comment:
Does this always wait for 2 mins?  If so, could we lower time to wait to 30s or 60s?

Reply:
I can change to 1m. The other two tests for the GC take 30-35s